libs/libpam/patches/020-backport-musl-patch.patch

```diff
@@ -0,0 +1,273 @@
+From c9593778a6133bf29eb2f47c24cc6d2f5d729fc8 Mon Sep 17 00:00:00 2001
+From: Fabrice Fontaine <fontaine.fabrice@gmail.com>
+Date: Thu, 11 Jun 2020 17:39:03 +0200
+Subject: [PATCH] Move check_user_in_passwd from pam_localuser.c to pam_modutil
+
+Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>
+
+* modules/pam_localuser/pam_localuser.c: Include
+<security/pam_modutil.h>.
+(pam_sm_authenticate): Replace check_user_in_passwd with
+pam_modutil_check_user_in_passwd.
+(check_user_in_passwd): Rename to pam_modutil_check_user_in_passwd,
+move to ...
+* libpam/pam_modutil_check_user.c: ... new file.
+* libpam/Makefile.am (libpam_la_SOURCES): Add pam_modutil_check_user.c.
+* libpam/include/security/pam_modutil.h
+(pam_modutil_check_user_in_passwd): New function declaration.
+* libpam/libpam.map (LIBPAM_MODUTIL_1.4.1): New interface.
+
+Co-authored-by: Dmitry V. Levin <ldv@altlinux.org>
+---
+ libpam/Makefile.am                    |  1 +
+ libpam/include/security/pam_modutil.h |  5 ++
+ libpam/libpam.map                     |  5 ++
+ libpam/pam_modutil_check_user.c       | 90 +++++++++++++++++++++++++++
+ modules/pam_localuser/pam_localuser.c | 86 +------------------------
+ 5 files changed, 103 insertions(+), 84 deletions(-)
+ create mode 100644 libpam/pam_modutil_check_user.c
+
+diff --git a/libpam/Makefile.am b/libpam/Makefile.am
+index 9252a837..11a1f329 100644
+--- a/libpam/Makefile.am
++++ b/libpam/Makefile.am
+@@ -35,6 +35,7 @@ libpam_la_SOURCES = pam_account.c pam_auth.c pam_data.c pam_delay.c \
+ 	pam_misc.c pam_password.c pam_prelude.c \
+ 	pam_session.c pam_start.c pam_strerror.c \
+ 	pam_vprompt.c pam_syslog.c pam_dynamic.c pam_audit.c \
++	pam_modutil_check_user.c \
+ 	pam_modutil_cleanup.c pam_modutil_getpwnam.c pam_modutil_ioloop.c \
+ 	pam_modutil_getgrgid.c pam_modutil_getpwuid.c pam_modutil_getgrnam.c \
+ 	pam_modutil_getspnam.c pam_modutil_getlogin.c pam_modutil_ingroup.c \
+diff --git a/libpam/include/security/pam_modutil.h b/libpam/include/security/pam_modutil.h
+index 3a6aec6a..33f87b90 100644
+--- a/libpam/include/security/pam_modutil.h
++++ b/libpam/include/security/pam_modutil.h
+@@ -58,6 +58,11 @@ extern "C" {
+ 
+ #include <security/_pam_types.h>
+ 
++extern int PAM_NONNULL((1,2))
++pam_modutil_check_user_in_passwd(pam_handle_t *pamh,
++                                 const char *user_name,
++                                 const char *file_name);
++
+ extern struct passwd * PAM_NONNULL((1,2))
+ pam_modutil_getpwnam(pam_handle_t *pamh, const char *user);
+ 
+diff --git a/libpam/libpam.map b/libpam/libpam.map
+index c9690a91..3cc7ef35 100644
+--- a/libpam/libpam.map
++++ b/libpam/libpam.map
+@@ -82,3 +82,8 @@ LIBPAM_1.4 {
+   global:
+     pam_start_confdir;
+ } LIBPAM_1.0;
++
++LIBPAM_MODUTIL_1.4.1 {
++  global:
++    pam_modutil_check_user_in_passwd;
```

Comment:
Is this patch necessary? It seems to be introducing a new API/ABI for the library which may not appear, or may appear modified in 1.4.1.

Reply:
Sort of. I figured that the simplest solution was to backport everything.